packages/app/src/cli/services/dev/app-events/app-watcher-esbuild.ts

```diff
@@ -31,35 +32,46 @@ export class ESBuildContextManager {
   setAbortSignal(signal: AbortSignal) {
     this.signal = signal
     this.signal?.addEventListener('abort', async () => {
-      const allDispose = Object.values(this.contexts).map((context) => context.dispose())
+      const allDispose = Object.values(this.contexts)
+        .map((context) => context.map((ctxt) => ctxt.dispose()))
+        .flat()
       await Promise.all(allDispose)
     })
   }
 
   async createContexts(extensions: ExtensionInstance[]) {
     const promises = extensions.map(async (extension) => {
-      const esbuildOptions = getESBuildOptions({
-        minify: false,
-        outputPath: extension.getOutputPathForDirectory(this.outputPath),
-        environment: 'development',
-        env: {
-          ...this.dotEnvVariables,
-          APP_URL: this.url,
-        },
-        stdin: {
-          contents: extension.getBundleExtensionStdinContent(),
+      const {main, assets} = extension.getBundleExtensionStdinContent()
+      const contexts: BuildContext[] = []
+      const mainOutputPath = extension.getOutputPathForDirectory(this.outputPath)
+      const esbuildOptions = await this.extensionEsBuildOptions(
+        {
+          contents: main,
           resolveDir: extension.directory,
           loader: 'tsx',
         },
-        logLevel: 'silent',
-        // stdout and stderr are mandatory, but not actually used
-        stderr: process.stderr,
-        stdout: process.stdout,
-        sourceMaps: true,
-      })
-
+        mainOutputPath,
+      )
       const context = await esContext(esbuildOptions)
-      this.contexts[extension.handle] = context
+      contexts.push(context)
+
+      if (assets) {
+        await Promise.all(
+          assets.map(async (asset) => {
+            const esbuildOptions = await this.extensionEsBuildOptions(
+              {
+                contents: asset.content,
+                resolveDir: extension.directory,
+                loader: 'ts',
```

Comment:
Is this intended to be `ts`? can it be anything else?

Reply:
Yes, for this use case we only need `ts`. But as a future improvement, this can be configurable from within the build_manifest object.